src/microbe_stage/editor/MicrobeEditorGUI.cs

```diff
@@ -1084,43 +1127,44 @@ internal void PlayInvalidActionSound()
     }
 
     /// <summary>
-    ///   Lock / unlock the organelles  that need a nuclues
+    ///   Lock / unlock the organelles that need a nucleus
     /// </summary>
-    /// <remarks>
-    ///   <para>
-    ///     TODO: rename to something more sensible
-    ///     and maybe also improve how this is implemented
-    ///     to be not cluttered
-    ///   </para>
-    /// </remarks>
-    internal void UpdateGuiButtonStatus(bool hasNucleus)
+    internal void UpdatePartsAvailability(bool hasNucleus)
     {
-        foreach (Control organelleItem in organelleSelectionElements)
+        foreach (var organelleItem in placeablePartSelectionElements)
         {
-            SetOrganelleButtonStatus(organelleItem, hasNucleus);
+            // organelleItem could be a non-implemented organelle so check if it really exist first.
+            if (!SimulationParameters.Instance.HasOrganelle(organelleItem.Name))
+            {
+                organelleItem.Locked = true;
+                continue;
+            }
+
+            if (organelleItem.Name == nucleus.InternalName)
+            {
+                organelleItem.Locked = hasNucleus;
+            }
+            else if (SimulationParameters.Instance.GetOrganelleType(organelleItem.Name).RequireNucleus)
+            {
+                organelleItem.Locked = !hasNucleus;
+            }
```

Comment:
I think that should be extracted to a `UpdatePartAvailability(bool hasNucleus, MicrobePartSelection organelleItem)` method for clearance